crates/bevy_ecs/src/query/state.rs

```diff
@@ -835,6 +837,8 @@ impl<Q: WorldQuery, F: ReadOnlyWorldQuery> QueryState<Q, F> {
         QueryParIter {
             world,
             state: self,
+            last_run: world.last_change_tick(),
+            this_run: world.read_change_tick(),
```

Comment:
We have access to `&mut World` here. We should be able to use non-atomic reads of `change_tick`.